### PR TITLE
docs: Add Kafka Schema Registry compliant JSON

### PR DESCRIPTION
**Pull request by muirandy**
docs: Add recipe for Kafka Schema Registry compliant JSON

### CODE REVIEW

File: website/docs/recipes/kafka.md [hunk 1]
```diff
@@ -169,3 +170,217 @@ private MessageAndMetadata createPactRepresentationFor(Map<String, Object> metad
...
+            FAIL_UNKNOWN_PROPERTIES,
+            new ObjectMapper(),
+            schemaRegistryClient
+    );
+}
+```
+We're using a MockSchemaRegistryClient which is provided out-of-the-box. Given that we're not actually running Kafka, we're also not running the Schema Registry, so we don't want anything to try and communicate with it, and thats what this class does for us. We need to register the subject (usually the Kafka topic + either "-key" or "-value") along with the JsonSchema itself into the MockSchemaRegistryClient.
+
+## The Pact
+As ever, running the consumer test will generate the Pact file (JSON), and push it out to the Pact Broker (well, you are using one, right?!):
```
Comment (YOU54F) [inline website/docs/recipes/kafka.md:288]: push it out to the Pact Broker if running in CI, or publishPacts flag is set?

Is it worth making this distinction, as generally you wouldn't publish pacts from local runs?

File: website/docs/recipes/kafka.md [hunk 2]
```diff
@@ -169,3 +170,217 @@ private MessageAndMetadata createPactRepresentationFor(Map<String, Object> metad
...
+    PactDslJsonBody body = new PactDslJsonBody();
+    body.stringType("name", "almost-anything2");
+
+    return builder.expectsToReceive("A json schema message")
+                  .withMetadata(Map.of("contentType", "application/vnd.schemaregistry.v1+json"))
+                  .withContent(body)
+                  .toPact();
+}
+```
+And there was our change: the metadata has been changed to match that of a schema registry compliant JSON message. This is significant - Pact will now be aware of the need for those 5 magic bytes mentioned earlier. We can now write our test. First, lets add a few constants and a field:
```
Comment (YOU54F) [inline website/docs/recipes/kafka.md:202]: would showing the snippet with and without the change, or having an annotated comment on the line, I assume it line 197 is the money ticket here from reading the paragraph on 202?

File: website/docs/recipes/kafka.md [hunk 3]
```diff
@@ -169,3 +170,217 @@ private MessageAndMetadata createPactRepresentationFor(Map<String, Object> metad
...
+  },
+  "provider": {
+    "name": "jsonSchemaKafkaProviderApp"
+  }
+}
+```
+Hopefully, that's pretty much what you expected to see. Again, note the contentType shows up.
+
+## Schema Registry JSON Provider
+Lets take a lok on the other side now - the provider test. We'll get our JUnit5 test class into being, along with a handful of constants:
```
Comment (YOU54F) [inline website/docs/recipes/kafka.md:334]: typo lok - look

**Review by YOU54F — APPROVED** [review]
Looks great buddy, couple of minor points, bar the typo should be good to go

Have you got an associated repo with your sample code? Always good to link from a blog post to some code on your repo, to get eyes on your code/GH profile 👍🏾

File: website/docs/recipes/kafka.md (hunk 1 above)
Comment (muirandy) [reply website/docs/recipes/kafka.md:288]: Yes, indeed... will fix!

File: website/docs/recipes/kafka.md (hunk 2 above)
Comment (muirandy) [reply website/docs/recipes/kafka.md:202]: Yes - I've given it a shot!

File: website/docs/recipes/kafka.md (hunk 3 above)
Comment (muirandy) [reply website/docs/recipes/kafka.md:334]: Yup!

**Comment by muirandy** [comment]
FYI, the code sits here:
https://github.com/muirandy/pact-kafka-schema-registry-demo

**Comment by muirandy** [comment]
Agh. Didn't mean to close the PR...

**Review by YOU54F — APPROVED** [review]
Thanks for the changes Andy, looks great. I don’t have write access yet but will get @mefellows or one of the team to give it a merge.

look forward to seeing and sharing your blog post ✌🏽

**Comment by YOU54F** [comment]
> FYI, the code sits here:
> https://github.com/muirandy/pact-kafka-schema-registry-demo

I’d link the repo in your blog post for sure, I always enjoyed reading blog posts and then taking a look at the code.

**Pull request merged**